graphqlapi/src/main/java/io/stargate/graphql/persistence/schemafirst/SchemaSourceDao.java

```diff
@@ -282,4 +286,33 @@ public void abortDeployment(String namespace) throws ExecutionException, Interru
             .bind();
     dataStore.execute(updateDeploymentToNotInProgress).get();
   }
+
+  public void purgeOldSchemaEntries(String namespace) throws Exception {
+    List<SchemaSource> allSchemasForNamespace = getSchemaHistory(namespace);
+
+    int numberOfEntriesToRemove =
+        allSchemasForNamespace.size() - NUMBER_OF_RETAINED_SCHEMA_VERSIONS;
+    if (numberOfEntriesToRemove > 0) {
+      logger.info("Removing {} old schema entries.", numberOfEntriesToRemove);
+
+      // remove N oldest entries
+      List<SchemaSource> schemasToRemove =
+          allSchemasForNamespace.subList(
+              NUMBER_OF_RETAINED_SCHEMA_VERSIONS, allSchemasForNamespace.size());
+
+      for (SchemaSource schemaSource : schemasToRemove) {
+        BoundQuery deleteSchemaQuery =
+            dataStore
+                .queryBuilder()
+                .delete()
+                .from(namespace, TABLE_NAME)
+                .where(KEY_CONDITION)
+                .where(VERSION_COLUMN_NAME, Predicate.EQ, schemaSource.getVersion())
+                .build()
+                .bind();
+
+        dataStore.execute(deleteSchemaQuery).get();
+      }
```

Comment:
This works, but since timeuuids are comparable it can be simplified to a single query:
```
delete from graphql_schema where key = 'key' and version < :oldest_version_to_keep;
```